.github/workflows/dependencies.yaml

```
@@ -0,0 +1,38 @@
+name: Update dependencies
+on:
+  schedule:
+    - cron: '0 3 * * 6'  # Run every saturday at 3 am.
+jobs:
+  update_dependencies:
+    name: Update dependencies
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip dir
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: pip-cache-${{ matrix.python-version }}-${{ hashFiles('requirements.txt') }}
+      - name: Update dependencies
+        run: | 
+          python -m pip install --upgrade pip setuptools poetry
+          python -m pip install -r lib/galaxy/dependencies/lint-requirements.txt
+          python -m pip freeze -l > lib/galaxy/dependencies/pinned-lint-requirements.txt
+          poetry update -v --lock
+          poetry export -f requirements.txt --without-hashes --output lib/galaxy/dependencies/pinned-requirements.txt
+          poetry export --dev -f requirements.txt --without-hashes --output lib/galaxy/dependencies/dev-requirements.txt
```

Can't you just re-use the make target?
```suggestion
          python -m venv .venv
          make update-dependencies
```

---

There was an issue with that. I'll try again - maybe it's as simple as creating a venv.

---

`./lib/galaxy/dependencies/update_lint_requirements.sh: 7: virtualenv: not found`
@nsoranzo  That script calls `virtualenv`. Do you think we can just use the venv module instead?

---

Yes, I think you can replace `virtualenv -p python3` with `python3 -m venv` in `lib/galaxy/dependencies/update_lint_requirements.sh`

---

I think it's just `python` (not `python3`) - we don't have Python 2 there. Trying it now..